### PR TITLE
feat: add local docker mode

### PR DESCRIPTION
**Pull request by gitlawr**
1. Support docker in local environment mode.
2. Add docker operator implementation
3. Support cli install/unistall local walrus

Examples:
```
$ walrus local install
$ walrus local uninstall
```

Related issue: https://github.com/seal-io/walrus/issues/1710

### CODE REVIEW

File: pkg/operator/docker/resourcestatus/status.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+// resourceTypes indicate supported resource type and function to get status.
+var resourceTypes map[string]getStatusFunc
+
+// getStatusFunc is function use resource id to get resource status.
+type getStatusFunc func(ctx context.Context, client *client.Client, resourceType, name string) (*status.Status, error)
+
+func init() {
+	resourceTypes = map[string]getStatusFunc{
+		"docker_container": getContainerStatus,
```
Comment (shanewxy) [inline pkg/operator/docker/resourcestatus/status.go:20]: Do we need to include other resource types like docker_image?

Comment (gitlawr) [reply pkg/operator/docker/resourcestatus/status.go:20]: We can make further enhancements based on requirements or what we provide in our built-in templates.

**Review by shanewxy — DISMISSED** [review]
LGTM

File: pkg/server/init_local_environment.go [hunk 2]
```diff
@@ -148,3 +185,73 @@ func (r *Server) readKubeConfig() (string, error) {
...
+		Where(
+			connector.Name("local"),
+			connector.ProjectID(projectID),
+		).
+		Only(ctx)
+
+	if model.IsNotFound(err) {
+		conn = &model.Connector{
+			Name:                      "local",
+			Type:                      "docker",
```
Comment (aiwantaozi) [inline pkg/server/init_local_environment.go:204]: use types.ConnectorCategoryDocker

File: pkg/terraform/convertor/docker.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/seal-io/walrus/pkg/dao/model"
+	"github.com/seal-io/walrus/pkg/dao/types/property"
+	"github.com/seal-io/walrus/pkg/terraform/block"
+	"github.com/seal-io/walrus/utils/log"
+)
+
+type DockerConvertor string
+
+func (m DockerConvertor) IsSupported(connector *model.Connector) bool {
+	return connector.Type == "docker"
```
Comment (aiwantaozi) [inline pkg/terraform/convertor/docker.go:15]: types.ConnectorCategoryDocker

File: pkg/cli/cmd/local.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/seal-io/walrus/pkg/cli/config"
+	"github.com/seal-io/walrus/pkg/cli/local"
+)
+
+// NewLocalCmd generate local command.
+func NewLocalCmd() *cobra.Command {
+	// Command install.
+	installCmd := &cobra.Command{
+		Use:   "install",
+		Short: "Install local Walrus setup",
```
Comment (aiwantaozi) [inline pkg/cli/cmd/local.go:22]: `Install local Walrus` is enough?

File: pkg/cli/cmd/local.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+			if err := install(); err != nil {
+				panic(err)
+			}
+		},
+	}
+
+	// Command uninstall.
+	uninstallCmd := &cobra.Command{
+		Use:   "uninstall",
+		Short: "Uninstall local Walrus setup",
```
Comment (aiwantaozi) [inline pkg/cli/cmd/local.go:33]: Ditto

File: pkg/cli/local/uninstall.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	cmd := exec.Command("docker", "extension", "rm", "sealio/walrus-docker-extension")
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("%w: %s", err, string(output))
+	}
+
+	return nil
+}
+
+func uninstallLocalWalrusDockerContainer() error {
+	cmd := exec.Command("docker", "rm", "--force", "local-walrus")
```
Comment (aiwantaozi) [inline pkg/cli/local/uninstall.go:33]: could you add const for and local-walrus and sealio/walrus-docker-extension

**Comment by gitlawr** [comment]
Updated.

**Review by aiwantaozi — APPROVED** [review]
LGTM

**Pull request merged**